tests/perf-system/analyzer/analyzer.py

```diff
@@ -1,348 +1,490 @@
+            bins.append(bin_val)
+
+        for lat in self.ms_latency_list:
+            counts[
+                int(
+                    (lat - min_latency) // ms_separator
+                    + bool((lat - min_latency) % ms_separator)
+                )
+            ] += 1
+
+        top_bins = []
+        top_counts = []
+        min_count = sorted(counts)[-highest_vals]
+        for ind in range(len(counts)):
+            if counts[ind] >= min_count:
+                top_bins.append(round(bins[ind], 3))
+                top_counts.append(counts[ind])
+
+        x_axis = range(len(top_bins))
+        plt.figure()
+        fig, ax = plt.subplots()
+        ax.bar(x_axis, top_counts, 0.9, align="center")
+        ax.set_xticks(x_axis)
+        ax.set_xticklabels(top_bins, rotation=25)
+        plt.ylabel("Count")
+        plt.xlabel("Latency (ms)")
+        fig.subplots_adjust(bottom=0.2)
+        plt.savefig(results_directory / "latency_distribution.pdf")
+        plt.close()
+
+
+def get_df_from_parquet_file(input_file: Path):
+    return pd.read_parquet(input_file)
+
+
+def prettify_graphs():
+    fontsize = 9
+    params = {
+        "axes.labelsize": fontsize,
+        "font.size": fontsize,
+        "legend.fontsize": fontsize,
+        "xtick.labelsize": fontsize,
+        "ytick.labelsize": fontsize,
+        "figure.figsize": DEFAULT_FIGURE_SIZE,
+        "xtick.direction": "in",
+        "ytick.direction": "in",
+        "xtick.top": True,
+        "ytick.right": True,
+        "xtick.minor.visible": True,
+        "ytick.minor.visible": True,
+        "lines.linewidth": 2,
+        "legend.frameon": False,
+        "axes.grid": False,
+        "savefig.bbox": "tight",
+    }
+    plt.rcParams.update(params)
+
+
+def default_analysis(
+    input_file: Path,
+    send_file: Path,
+    response_file: Path,
+    results_directory=Path("."),
+    error_on_failure=False,
+    plot_graphs=True,
+):
+    """
+    Produce the analysis results.
+    A summary of results will be printed to stdout and a collection of
+    graphs will also be saved to results_directory, provided that plot_graphs
+    is not False. Anaylsis will termination with an error if error_on_failure
+    is set and any requests have not succeeded.
+    """
+    analysis = Analyze()
+    df_input = get_df_from_parquet_file(input_file)
+    df_sends = get_df_from_parquet_file(send_file)
+    df_responses = get_df_from_parquet_file(response_file)
+
+    successful_percent = analysis.iter_for_success_and_latency(
+        df_input, df_sends, df_responses
+    )
+
+    if error_on_failure and successful_percent < 100:
+        assert False, "Not all requests were successful"
+
+    LOG.info(f"The request type sent is {analysis.get_req_type(df_responses)}")
+    LOG.info(
+        f"Request verbs counts are: POST {analysis.request_verb.count('POST')}, GET {analysis.request_verb.count('GET')}, DELETE {analysis.request_verb.count('DELETE')}"
+    )
```

Comment:
```suggestion
    verbs = analysis.request_verb
    verb_counts = {v: verbs.count(v) for v in set(verbs)}
    LOG.info(
        f"Request verbs counts are: { ', '.join(f'{v} {c}' for v,c in verb_counts.items()) }"
    )
```

Something like that, but please test my GitHub Python...